File: Rust/src/bin/day1-2.rs
fn match_numwrd(numwrd: &str) -> Option<usize>{
    let wrdidx = numwrd.len();
    if numwrd.find("one")? < wrdidx{ Some(1) }
    else if numwrd.find("two")? < wrdidx{ Some(2) }
    else if numwrd.find("three")? < wrdidx{ Some(3) }
    else if numwrd.find("four")? < wrdidx{ Some(4) }
    else if numwrd.find("five")? < wrdidx{ Some(5) }
    else if numwrd.find("six")? < wrdidx{ Some(6) }
    else if numwrd.find("seven")? < wrdidx{ Some(7) }
    else if numwrd.find("eight")? < wrdidx{ Some(8) }
    else if numwrd.find("nine")? < wrdidx{ Some(9) }
    else {None}
}

fn rmatch_numwrd(numwrd: &str) -> Option<usize>{
    let wrdidx = 0;
    if numwrd.find("one")? < wrdidx{ Some(1) }
    else if numwrd.rfind("two")? > wrdidx{ Some(2) }
    else if numwrd.rfind("three")? > wrdidx{ Some(3) }
    else if numwrd.rfind("four")? > wrdidx{ Some(4) }
    else if numwrd.rfind("five")? > wrdidx{ Some(5) }
    else if numwrd.rfind("six")? > wrdidx{ Some(6) }
    else if numwrd.rfind("seven")? > wrdidx{ Some(7) }
    else if numwrd.rfind("eight")? > wrdidx{ Some(8) }
    else if numwrd.rfind("nine")? > wrdidx{ Some(9) }
    else {None}
}

fn get_lhs(buffer: &String) -> Option<char> {
    let idx;
    let num_idx = buffer.find(char::is_numeric).unwrap();
    let wrd_idx = match_numwrd(buffer.as_str());
    if wrd_idx.is_none(){idx = num_idx;}
    else {idx = std::cmp::min(wrd_idx?, num_idx);}
    Some(buffer.chars().nth(idx).unwrap())
}

fn get_rhs(buffer: &String) -> Option<char> {
    let idx;
    let num_idx = buffer.rfind(char::is_numeric).unwrap();
    let wrd_idx = rmatch_numwrd(buffer.as_str());
    if wrd_idx.is_none(){idx = num_idx;}
    else {idx = std::cmp::max(wrd_idx?, num_idx);}
    Some(buffer.chars().nth(idx).unwrap())
}

fn main() -> Result<(), std::io::ErrorKind> {
    let mut total: i64 = 0;
    let mut calval = String::with_capacity(2);
    for line in std::io::stdin().lines() {
        let buffer = line.expect("Shit");
        calval.push(get_lhs(&buffer).ok_or(std::io::ErrorKind::InvalidData)?);
        calval.push(get_rhs(&buffer).ok_or(std::io::ErrorKind::InvalidData)?);
        println!("{}", calval);
        total += calval.parse::<i64>().unwrap();
        calval.clear();
    }
    println!("Total: {}", total);
    return Ok(())
}

File: C++/day1-2.cpp
#include <iostream>
#include <array>
#include <string>
#include <algorithm>

static const char ascii_num_lo = 48;
static const char ascii_num_hi = 57;
static const std::array<std::string, 9> numwrds = {"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

bool is_numeric(const char character){
    return ascii_num_lo <= character && character <= ascii_num_hi;
}

bool are_words_in(std::string full_string){
    return std::any_of(numwrds.begin(), numwrds.end(),
            [&](const std::string str){return full_string.find(str) != std::string::npos;});
}

std::string tokenise(std::string orig_string){
    std::string new_string;
    new_string.append();
    return new_string;
}

int main(){
    int total = 0;
    int this_line = 0;
    for (std::string line; std::getline(std::cin, line);){
        this_line = 0;
        std::cout << "Line: " << line << "\t\tNumber: " << this_line << "\n";
        total += this_line;
    }
    std::cout << "Final Total: " << total << std::endl;
}
